execute/dataset.go

```diff
@@ -234,3 +236,97 @@ func (d *PassthroughDataset) Finish(err error) {
 
 func (d *PassthroughDataset) SetTriggerSpec(t plan.TriggerSpec) {
 }
+
+// TransportDataset holds data for a specific node and sends
+// messages to downstream nodes using the Transport.
+//
+// This Dataset also implements a shim for execute.Dataset
+// so it can be integrated with the existing execution engine.
+// These methods are stubs and do not do anything.
+type TransportDataset struct {
+	id    DatasetID
+	ts    []Transport
```

Comment:
Could we call this `transports`? I'm kinda starting to thing we should be better about using better names.